session_test.go

```diff
@@ -772,6 +772,17 @@ var _ = Describe("Session", func() {
 			Expect(mconn.written[0][0] & 0x02).ToNot(BeZero()) // Public Reset
 			Expect(sess.runClosed).To(BeClosed())
 		})
+
+		It("unblocks WaitUntilClosed when the run loop exists", func() {
```

Comment:
We're making our tests race-free? That'll be a long way to go and a lot of bools to change.